marshalers.go

```diff
@@ -14,6 +14,10 @@ import (
 )
 
 func marshalPtr(data interface{}, length int) (internal.Pointer, error) {
+	if length == 0 {
```

Comment:
I think I'd prefer to check data being nil, since this is what the user passes. Maybe something like the following?

```
	if data == nil {
		if length != 0 {
			return internal.Pointer{}, errors.New("...")
```

Currently doing `Lookup(&garbage, &v)` doesn't return an error.